xbee_android_library/src/main/java/com/digi/xbee/api/android/XBeeDevice.java

```diff
@@ -0,0 +1,174 @@
+     * @see #XBeeDevice(Context, String, int)
+     * @see #XBeeDevice(Context, String, SerialPortParameters)
+     * @see com.digi.xbee.api.android.connection.android.AndroidUSBPermissionListener
+     */
+    public XBeeDevice(Context context, int baudRate, AndroidUSBPermissionListener permissionListener) {
+        super(XBee.createConnectiontionInterface(context, baudRate, permissionListener));
+    }
+
+    /**
+     * Class constructor. Instantiates a new {@code XBeeDevice} object for
+     * Android with the given parameters.
+     *
+     * <p>This constructor uses the Digi Android Serial Port API based on the
+     * RxTx library to communicate with the devices.</p>
+     *
+     * @param context The Android application context.
+     * @param port Serial port name where XBee device is attached to.
+     * @param baudRate The serial port connection baud rate.
+     *
+     * @throws NullPointerException If {@code context == null} or
+     *                              if {@code port == null}.
+     * @throws IllegalArgumentException if {@code baudRate < 1}.
+     *
+     * @see #XBeeDevice(Context, int)
+     * @see #XBeeDevice(Context, int, AndroidUSBPermissionListener)
+     * @see #XBeeDevice(Context, String, SerialPortParameters)
+     */
+    public XBeeDevice(Context context, String port, int baudRate) {
+        super(XBee.createConnectiontionInterface(context, port, baudRate));
+    }
+
+    /**
+     * Class constructor. Instantiates a new {@code XBeeDevice} object for
+     * Android with the given parameters.
+     *
+     * <p>This constructor uses the Digi Android Serial Port API based on the
+     * RxTx library to communicate with the devices.</p>
+     *
+     * @param context The Android application context.
+     * @param port Serial port name where XBee device is attached to.
+     * @param parameters The serial port parameters.
+     *
+     * @throws NullPointerException If {@code context == null} or
+     *                              if {@code port == null} or
+     *                              if {@code parameters == null}.
+     *
+     * @see #XBeeDevice(Context, int)
+     * @see #XBeeDevice(Context, int, AndroidUSBPermissionListener)
+     * @see #XBeeDevice(Context, String, int)
+     * @see com.digi.xbee.api.connection.serial.SerialPortParameters
+     */
+    public XBeeDevice(Context context, String port, SerialPortParameters parameters) {
+        super(XBee.createConnectiontionInterface(context, port, parameters));
+    }
+
+    /**
+     * Class constructor. Instantiates a new {@code XBeeDevice} object for
+     * Android with the given parameters.
+     *
+     * <p>This constructor uses the Android Bluetooth Low Energy API to
+     * communicate with the devices.</p>
+     *
+     * <p>The Bluetooth password must be provided before calling the
+     * {@link #open()} method, either through this constructor or the
+     * {@link #setBluetoothPassword(String)} method.</p>
```

Comment:
I would change the visibility of the `setBluetoothPassword(String)` method to `protected` in the Java library and make it public in these subclasses.

Reply:
Will do.